### PR TITLE
LPS-180735 | LPS-180736 Crawler user agents and use it for Layout SEO dynamic Rendering

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
https://issues.liferay.com/browse/LPS-180049

On this PRs, we will use the configuration for the user agents that now is alone for the previous LayoutSEODynamicRenderingConfiguration

image

image

- LPS-180735 show the new configuration
- LPS-180736 update the value of the new configuration with the existing values of the LayoutSEODynamicRenderingConfiguration
- LPS-180736 add test to test the upgrade
- LPS-180736 use the method from the userAgentMatcher to know if the user agent is on the list of the crawler user agents
- LPS-180736 remove the crawlerUserAgents properties from the layout seo rendering configuration
- LPS-180736 put message to let the user know where the crawler user agent configuration is now
- LPS-180736 buildLang

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/servlet/taglib/LayoutSEODynamicRenderingTopHeadJSPDynamicInclude.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		String configurationPid =
+			CrawlerUserAgentsConfiguration.class.getName();
+
+		httpServletRequest.setAttribute(
+			"edit-configuration.jsp-configurationURL",
+			_language.format(
+				httpServletRequest,
+				"to-set-up-the-user-agents-go-to-crawler-user-agents",
+				new String[] {
+					"<a href=\"" +
```
Comment (AliciaGarciaGarcia) [inline modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/servlet/taglib/LayoutSEODynamicRenderingTopHeadJSPDynamicInclude.java:66]: @ambrinchaudhary 

This way to build the <a ... > is ok, there is something missing for the accesibility?

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/servlet/taglib/LayoutSEODynamicRenderingTopHeadJSPDynamicInclude.java (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/servlet/taglib/LayoutSEODynamicRenderingTopHeadJSPDynamicInclude.java:66]: Hey @AliciaGarciaGarcia, this link has no text associated. Usually links look like `<a href="...">click here</a>`. 
But I'm not sure if that is what you need... if you are just setting a request attribute, why don't you simply set the link, without any `<a> `tag?

**Review by ambrinchaudhary — APPROVED** [review]

File: modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/servlet/taglib/LayoutSEODynamicRenderingTopHeadJSPDynamicInclude.java (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/servlet/taglib/LayoutSEODynamicRenderingTopHeadJSPDynamicInclude.java:66]: Talked about this via slack. What you have is enough from accessibility point of view

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
I do not think that [LocalFile.WebContentDisplayWithStaging#ViewWebContentWithEditedImageField](https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/557671//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_WebContentDisplayWithStaging_ViewWebContentWithEditedImageField_) is related with this PR.

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by WYZ-Issac** [comment]
The above failures are not related to this pull. I will forward this directly.

**Comment by WYZ-Issac** [comment]
ci:forward:force

**Pull request closed without merging**